crates/miden-objects/src/account/component/template/storage/entry_content.rs

```diff
@@ -466,67 +466,113 @@ impl Deserializable for FeltRepresentation {
 /// Supported map representations for a component's storage entries.
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[cfg_attr(feature = "std", derive(::serde::Deserialize, ::serde::Serialize))]
-pub struct MapRepresentation {
-    /// The human-readable name of the map slot.
-    /// An optional description for the slot, explaining its purpose.
-    identifier: FieldIdentifier,
-    /// Storage map entries, consisting of a list of keys associated with their values.
-    entries: Vec<MapEntry>,
+pub enum MapRepresentation {
+    /// A map whose contents are provided during instantiation via placeholders.
+    Template {
+        /// The human-readable identifier of the map slot.
+        identifier: FieldIdentifier,
+    },
+    /// A map with statically defined key/value pairs.
+    Value {
+        /// The human-readable identifier of the map slot.
+        identifier: FieldIdentifier,
+        /// Storage map entries, consisting of a list of keys associated with their values.
+        entries: Vec<MapEntry>,
+    },
 }
 
 impl MapRepresentation {
     /// Creates a new `MapRepresentation` from a vector of map entries.
     pub fn new(entries: Vec<MapEntry>, name: impl Into<StorageValueName>) -> Self {
```

Comment:
I'd also rename this constructor to match the variant name, so we have `new_template` and `new_value` overall.